### PR TITLE
[BOLT][merge-fdata]Fix support for fdata files starting with no_lbr on ARM platform 

### PR DESCRIPTION
**Pull request by zrxxzz**
The merge-fdata tool did not support fdata files starting with no_lbr, which was not user-friendly for the ARM platform.
Therefore, I fixed this issue to enable its use on the ARM platform (with no_lbr). Additionally, I validated the merging of samples across both LBR and Non-LBR scenarios.

### CODE REVIEW

**Comment by zrxxzz** [comment]
Through testing, it has been proven that even without LBR, merging samples on the ARM platform still achieves good results for MySQL. 
Whether based on the standard version or the PGO-optimized version, the baseline version sees an improvement of around 10%, and the PGO version sees an improvement of around 5%.

**Comment by zrxxzz** [comment]
Hello, @maksfb could you please take a look at this pull request? Thank you.

File: bolt/tools/merge-fdata/merge-fdata.cpp [hunk 1]
```diff
@@ -34,45 +34,32 @@ cl::OptionCategory MergeFdataCategory("merge-fdata options");
...
-  cl::init(false),
-  cl::Optional,
-  cl::cat(MergeFdataCategory));
-
-static cl::opt<std::string>
-OutputFilePath("o",
-  cl::value_desc("file"),
-  cl::desc("Write output to <file>"),
-  cl::cat(MergeFdataCategory));
+    InputDataFilenames(cl::Positional, cl::CommaSeparated,
```
Comment (aaupov) [inline bolt/tools/merge-fdata/merge-fdata.cpp:42]: Any particular reason to change the formatting? Did you change any options?

File: bolt/tools/merge-fdata/merge-fdata.cpp [hunk 2]
```diff
@@ -34,45 +34,32 @@ cl::OptionCategory MergeFdataCategory("merge-fdata options");
 
 enum SortType : char {
   ST_NONE,
-  ST_EXEC_COUNT,      /// Sort based on function execution count.
-  ST_TOTAL_BRANCHES,  /// Sort based on all branches in the function.
+  ST_EXEC_COUNT,     /// Sort based on function execution count.
```
Comment (aaupov) [inline bolt/tools/merge-fdata/merge-fdata.cpp:37]: Please keep formatting as is here

File: bolt/tools/merge-fdata/merge-fdata.cpp [hunk 3]
```diff
@@ -265,7 +252,10 @@ bool isYAML(const StringRef Filename) {
 void mergeLegacyProfiles(const SmallVectorImpl<std::string> &Filenames) {
   errs() << "Using legacy profile format.\n";
   std::optional<bool> BoltedCollection;
+  std::optional<bool> NoLbr;
   std::mutex BoltedCollectionMutex;
+  std::mutex NoLbrMutex;
```
Comment (aaupov) [inline bolt/tools/merge-fdata/merge-fdata.cpp:257]: Can we avoid the use of mutexes here by just setting bolted_collection/no_lbr per profile and checking that after we parse all profiles?

File: bolt/tools/merge-fdata/merge-fdata.cpp [hunk 4]
```diff
@@ -336,6 +345,8 @@ void mergeLegacyProfiles(const SmallVectorImpl<std::string> &Filenames) {
 
   if (BoltedCollection.value_or(false))
     output() << "boltedcollection\n";
+  if (NoLbr.value_or(false))
+    output() << NoLbrLabel << "\n";
```
Comment (aaupov) [inline bolt/tools/merge-fdata/merge-fdata.cpp:349]:
```suggestion
    output() << "no_lbr\n";
```

**Comment by maksfb** [comment]
> Hello, @maksfb could you please take a look at this pull request? Thank you.

Thanks for your PR! Looks good, but please address the comments above.

**Review by paschalis-mpeis — COMMENTED** [review]
@zrxxzz , @aaupov: given that #118481 will be merged soon, would it be OK if I close this one?

**Comment by aaupov** [comment]
> @zrxxzz , @aaupov: given that #118481 will be merged soon, would it be OK if I close this one?

I don't mind. Thanks @zrxxzz for the initial implementation.

**Comment by zrxxzz** [comment]
> @zrxxzz , @aaupov: given that #118481 will be merged soon, would it be OK if I close this one?
@paschalis-mpeis : OK，I don't mind

**Comment by zrxxzz** [comment]
> @zrxxzz , @aaupov: given that #118481 will be merged soon, would it be OK if I close this one?

ok, I don‘t mind

**Pull request closed without merging**